File: exercises/python_playwright/alerts/alerts_playwright.py
from playwright.sync_api import Dialog, Page


class AlertsPage:
    def __init__(self, page: Page) -> None:
        """Initialize the page object with Playwright locators."""
        self.page = page
        self.alert_button = page.get_by_role("button", name="Alert")
        self.confirm_button = page.get_by_role("button", name="Confirm")
        self.prompt_button = page.get_by_role("button", name="Prompt")

    def open(self) -> None:
        """Navigate to the Alerts exercise page."""
        self.page.goto("http://uitestingplayground.com/alerts")

    def click_alert(self) -> None:
        """Accept the alert dialog triggered by the Alert button."""
        self.page.once("dialog", lambda dialog: dialog.accept())
        self.alert_button.click()

    def click_confirm(self, accept: bool = True) -> str:
        """Handle the confirm dialog and return the follow-up alert message."""
        self.page.once("dialog", lambda dialog: dialog.accept() if accept else dialog.dismiss())
        self.confirm_button.click()

        second_alert_box: Dialog = self.page.wait_for_event("dialog", timeout=5000)
        message = second_alert_box.message
        second_alert_box.accept()
        return message

    def click_prompt(self, input_text: str) -> str:
        """Submit prompt input and return the follow-up alert message."""
        self.page.once("dialog", lambda dialog: dialog.accept(input_text))
        self.prompt_button.click()

        second_alert_box: Dialog = self.page.wait_for_event("dialog", timeout=5000)
        message = second_alert_box.message
        second_alert_box.accept()
        return message

File: exercises/javascript_playwright/alerts/alertsPlaywright.js
export class AlertsPage {
    /**
     *
     * @param {import(@playwright).Page} page
     */
    constructor(page) {
        this.page = page;
        this.alertButton = page.locator("#alertButton");
        this.confirmButton = page.locator("#confirmButton");
        this.promptButton = page.locator("#promptButton");
    }
    async openPage() {
        await this.page.goto("http://uitestingplayground.com/alerts");
    }

    async clickAlertButton() {
        await this.page.once("dialog", (dialog) => dialog.accept());
        await this.alertButton.click();
    }

    async clickConfirmButton(decision) {
        await this.page.once("dialog", (dialog) => {
            decision ? dialog.accept() : dialog.dismiss();
        });
        await this.confirmButton.click();
    }

    async clickPromptButton(input) {
        await this.page.once("dialog", (dialog) => dialog.accept(input));
        await this.promptButton.click();
    }

    async getFollowupAlert() {
        const dialog = await this.page.waitForEvent("dialog", {
            timeout: 5000,
        });
        const alertMessage = dialog.message();
        dialog.accept();

        return alertMessage;
    }
}
